### ISSUE TITLE
Netflow import

### ISSUE THREAD

**Issue by WilliamSalusky** [arkime/arkime#1617]
Support for import of externally sourced flow data into Arkime so that it can be factored into network investigations and visualizations. Ideally support would be flexible in allowing an administrator to configure support for alternate flow export formats (v5, v9, flexible, sFlow, csv, etc...). This should be a script that reads the files and creates sessions2 records.

More ideas from @philhagen:
* receive netflow 5, 7, 9; ipfix; sflow
* index alongside SPI data with the limited fields available (so FPC-originated SPI can be queried at the same time as NF-originated SPI)
* de-duplicate FPC-originated and NF-originated SPI (i suspect this is going to be a bit tricky because timestamps will not be precisely identical)
* de-duplicate NF-originated records seen at multiple exporters (even more tricky due to timing AND possibly byte count disparities)
* probably tag or otherwise indicate that a record is NF-originated
* ability to configure multiple-to-numerous exporters from the config file and/or UI (dozens to hundreds)

**Comment by tweemeterjop** [comment]
+1! I would really like this option see implemented.

**Comment by StammesOpfer** [comment]
Ideally this is not strictly a file import but serve as a NetFlow collector as well. 
If a collector is not in the cards can we get a watch folder option at least.
I've been dreaming of this being built-in for a long time (https://github.com/arkime/arkime/issues/1160)

**Comment by philhagen** [comment]
I'd say live collector is the only way to go - there are countless formats of at-rest storage of NetFlow and choosing any one (or ten) will just leave too many people without the ability to use it.  A live collector means just a handful of formats to cover.

One additional requirement ported over from Slack:
- stitch together the typically unidirectional flow records into a session-based accounting.

**Comment by tweemeterjop** [comment]
> I'd say live collector is the only way to go - there are countless formats of at-rest storage of NetFlow and choosing any one (or ten) will just leave too many people without the ability to use it. A live collector means just a handful of formats to cover.

I disagree. We use Arkime to analyse pre recorded (historic) network traffic both from wiretaps or netflow from switches where we cannot get an Arkime monitor.

So for our use case we rely on the import functionality of Netflow data.

**Comment by philhagen** [comment]
I guess the question would then become which of the dozens/hundreds of formats should be supported by the importer?

**Comment by StammesOpfer** [comment]
> > I'd say live collector is the only way to go - there are countless formats of at-rest storage of NetFlow and choosing any one (or ten) will just leave too many people without the ability to use it. A live collector means just a handful of formats to cover.
> 
> I disagree. We use Arkime to analyse pre recorded (historic) network traffic both from wiretaps or netflow from switches where we cannot get an Arkime monitor.
> 
> So for our use case we rely on the import functionality of Netflow data.

Would using a separate tool (e.g. nfreplay/flow-to-if) be a viable option to replay files into a collector? Then Arkime does not have to support all the storage formats.

**Comment by philhagen** [comment]
I think the replay path is a great one! That puts the onus on the collector software/platform to handle their own format(s).  I could also see a path using a direct conversion script that reads the existing files and restructures records to match Arline's schema, inserting them directly to the OS/ES index.

**Comment by awick** [comment]
Won't make everyone happy but #3674 is what Opus suggested :)
